### PR TITLE
feat(docs): Add a table-of-contents to the `docs` directory.

### LINKED ISSUE #869: Make a nice table of contents for the `docs` directory
As suggested here: https://github.com/mozilla/application-services/pull/826#discussion_r267758738

### PR DESCRIPTION
**Pull request by rfk**
Fixes #869. [Rendered](https://github.com/mozilla/application-services/blob/docs-table-of-contents/docs/README.md).

### CODE REVIEW

File: docs/build-and-publish-pipeline.md [hunk 1]
```diff
@@ -27,11 +27,7 @@ The key points:
     * [TaskCluster](../automation/taskcluster/README.md) is used to:
         * Build an Android binary release.
         * Upload Android library symbols to [Socorro](https://wiki.mozilla.org/Socorro).
-        * Publish it to the 'mozilla-appservices' organization on [bintray](https://bintray.com/),
-          which mirrors it to [jcenter](https://bintray.com/bintray/jcenter).
-           * (although this will soon change to publish to https://maven.mozilla.org).
-    * There is also a manual step where we mirror artifacts from bintray to maven.mozilla.org,
-      as a temporary measure until we can get automatic publishing set up correctly.
+        * Publish it to the [maven.mozilla.org](https://maven.mozilla.org).
```
Comment (rfk) [inline docs/build-and-publish-pipeline.md]: I updated the docs around bintray/maven while I was in here - @eoger r?

File: docs/howtos/consuming-rust-components-on-ios.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ The package is published as a release on github: https://github.com/mozilla/appl
 
 - Add the dependency line to the Cartfile, for instance: `github "mozilla/application-services" ~> "v0.16.1"` 
 - `carthage` will download MozillaAppServices.frameworks.zip, and add all the available frameworks to the 'Carthage/' dir.
-- Choose which framework to link against for your project (in the *Link Binary with Libraries* step in your Xcode target).
+- Link against the provided MozillaAppServices.framework (in the *Link Binary with Libraries* step in your Xcode target).
```
Comment (rfk) [inline docs/howtos/consuming-rust-components-on-ios.md]: I believe this needs to change to reflect the fact that we're no longer publishing individual component frameworks on iOS.

File: docs/howtos/signing-android-applications.md [hunk 3]
```diff
@@ -1,36 +0,0 @@
-> Android requires that all APKs be digitally signed with a certificate before they can be installed.
```
Comment (rfk) [inline docs/howtos/signing-android-applications.md]: Maybe controversial, but...I don't think this doc belongs here. It's specifically about generating apks, which we don't do in this repo, so it wasn't obvious how to link to it from the ToC.

I wonder if there's some more android-product-development-centric place we could move it to?

Comment (rfk) [reply docs/howtos/signing-android-applications.md]: Perhaps this could live in https://github.com/mozilla-mobile/shared-docs?  @vladikoff thoughts?

Comment (rfk) [reply docs/howtos/signing-android-applications.md]: Ref https://github.com/mozilla-mobile/shared-docs/issues/80 for following

Comment (vladikoff) [reply docs/howtos/signing-android-applications.md]: Ah this doc was useful before, we can move it though, it's okay 👍

Comment (rfk) [reply docs/howtos/signing-android-applications.md]: Yep, I don't think we should lose it entirely, it's 👍 content

**Pull request merged**